server/src/useCases/report/productCompatibility/handleGetProductRequest.js

```diff
@@ -9,7 +11,15 @@ module.exports = class ReportGetProductRequestHandler {
   }
 
   async getReports(repository) {
-    repository.aggregateByProduct({}, (err, result) => {
+    const { limit, offset } = this.req.query;
+
+    let validatedLimit;
+    validatedLimit = FiltersValidation.isFilterValid(limit) ? limit : null;
+    validatedLimit = FiltersValidation.maxLimitCheck(validatedLimit);
```

Comment:
What is behavior in case of validatedLimit===null?
Also I think we can refactor this validation. I think there's no need to have single function for offset and limit. In addition FiltersValidation can be renamed to be more descriptive (those filters are specific to pagination). Also with that much code I don't like putting this lines inside of request handler directly. It can be tricky to manage with more handlers. Can we create separate middleware function that would be attached to  routing for endpoints with pagination? This way we can ensure consistency.

Reply:
I added middleware to handle pagination filters, please let me know if that is what you meant @dborowiecki